test/xds/xds_client_integration_test.go

```diff
@@ -106,3 +107,43 @@ func (s) TestClientSideXDS(t *testing.T) {
 		t.Fatalf("rpc EmptyCall() failed: %v", err)
 	}
 }
+
+func (s) TestOutlierDetection(t *testing.T) {
+	oldOD := envconfig.XDSOutlierDetection
+	envconfig.XDSOutlierDetection = true
+	defer func() {
+		envconfig.XDSOutlierDetection = oldOD
+	}()
+
+	managementServer, nodeID, _, resolver, cleanup1 := e2e.SetupManagementServer(t)
+	defer cleanup1()
+
+	port, cleanup2 := startTestService(t, nil)
+	defer cleanup2()
+
+	const serviceName = "my-service-client-side-xds"
+	resources := e2e.DefaultClientResources(e2e.ResourceParams{
+		DialTarget: serviceName,
+		NodeID:     nodeID,
+		Host:       "localhost",
+		Port:       port,
+		SecLevel:   e2e.SecurityLevelNone,
+	})
```

Comment:
Do these resource even contain appropriate outlier detection configuration?

Reply:
It does. xdsclient will emit a nil Outlier Detection struct, which will cause the cds balancer to prepare a "no-op" Outlier Detection config. I verified it was present in the system manually through logs. Let me know if you want an explicit check.